File: Code/Prolog/fourteen/schulz_luke14/question1.cpp
// Luke Schulz
// 4/25/19
// ASU ID# = 1211984746
// 240 Tu-Th 4:30

#include <string>
#include <iostream>

using namespace std;


int foo(int x, int y){ // size n problem

    if (y<=0){ // stopping condition #1
        return x;
    }
    if( x<= 0){ // stopping condition #2
        return y;
    }

    if(x>=y){ // size m solution
        return ( x + foo(x-1 , y - 2));
    }
    if( x < y){// size m solution 2
        return ( y + foo(x-2, y-3));
    }
}

int main(){
    cout << "foo(5 , 6 ) = " ;
    cout << foo(5,6);

} // end main

File: code/Prolog/chen_240/fourteen/question1.pl
/* Luke Schulz, HW 14 ! */



/* BEGIN RULES */
foo(ANSWER , X , Y):-
	X =< 0 , ANSWER is Y , !.
foo(ANSWER , X , Y):-
	Y=< 0 , ANSWER is X , !.
foo(ANSWER , X , Y):- 
	X >= Y ,
	X2 is X - 1 ,
	Y2 is Y - 2 ,
	foo(A2 , X2, Y2), /*this will set A2 */
	ANSWER is X + A2, !.
foo(ANSWER , X , Y):-
	X < Y ,
	X2 is X - 2 ,
	Y2 is Y - 3 ,
	foo(A2 , X2, Y2) ,
	ANSWER is Y + A2 , !.



q1 :- foo(A , 5 , 6) ,
	write('The Answer of foo(A , 5 , 6) is ') ,
	write(A) .

